### PR TITLE
[DOC] Update changelog

### PR DESCRIPTION
**Pull request by eseiler**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -82,108 +83,158 @@ Note that 3.1.0 will be the first API stable release and interfaces in this rele
...
+#include <seqan3/{alignment/pairwise => core/algorithm}/alignment_range.hpp>
+```
+
+#### Argument Parser
+
+Header Changes:
+
+```cpp
+#include <seqan3/{core => argument_parser}/detail/terminal.hpp>
+```
```
Comment (marehr) [inline CHANGELOG.md:137]:
```suggestion
```

(upps it's detail code)

File: CHANGELOG.md [hunk 2]
```diff
@@ -82,108 +83,158 @@ Note that 3.1.0 will be the first API stable release and interfaces in this rele
...
 * The `seqan3::begin()`, `seqan3::end()`, `seqan3::cbegin()`, `seqan3::cend()`, `seqan3::size()`, `seqan3::empty()`
-  functions have been deprecated:
-  Use `std::ranges::{begin|end|cbegin|cend|size|empty}()` instead ([\#1663](https://github.com/seqan/seqan3/pull/1663)).
-* We removed `seqan3::forward_range`. Use `std::ranges::borrowed_range` instead ([\#2038](https://github.com/seqan/seqan3/pull/2038)).
-* Added `seqan3::views::minimiser`, a view that computes the minimum in a window shifted over a range of comparable values.
-  ([\#1654](https://github.com/seqan/seqan3/pull/1654)).
-* Added `seqan3::views::minimiser_hash`, a view that computes the minimisers of a range of type seqan3::semialphabet.
-  ([\#1721](https://https://github.com/seqan/seqan3/pull/1721)).
-* `seqan3::views:trim` has been renamed to `seqan3::views:trim_quality`.
+  functions have been deprecated.  Use `std::ranges::{begin|end|cbegin|cend|size|empty}()` instead
```
Comment (marehr) [inline CHANGELOG.md:151]:
```suggestion
  functions have been deprecated. Use `std::ranges::{begin|end|cbegin|cend|size|empty}()` instead
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -82,108 +83,158 @@ Note that 3.1.0 will be the first API stable release and interfaces in this rele
...
+  ([\#1663](https://github.com/seqan/seqan3/pull/1663)).
+* The `seqan3::forward_range` has been removed. Use `std::ranges::borrowed_range` instead
+  ([\#2038](https://github.com/seqan/seqan3/pull/2038)).
+* The `seqan3::views::minimiser` has been added. This is a view that computes the minimum in a window shifted over a
+  range of comparable values ([\#1654](https://github.com/seqan/seqan3/pull/1654)).
+* The `seqan3::views::minimiser_hash` has been added. This is a view that computes the minimisers of a range of type
+  `seqan3::semialphabet` ([\#1721](https://https://github.com/seqan/seqan3/pull/1721)).
```
Comment (marehr) [inline CHANGELOG.md:158]: Aren't those new features?

File: CHANGELOG.md [hunk 4]
```diff
@@ -82,108 +83,158 @@ Note that 3.1.0 will be the first API stable release and interfaces in this rele
...
-### Alignment
+#### Alignment
 
 * When invoking the alignment algorithm with a user defined thread count using the `seqan3::align_cfg::parallel`
-  configuration element, `std::thread::hardware_concurrency()` many threads were always spawned. This is now fixed and
+  configuration element, all available threads were used. This is now fixed and
   only the specified number of threads will be spawned ([\#1854](https://github.com/seqan/seqan3/pull/1854)).
-* Using an unsigned `score_type` is prevented with a static assert, since gaps and mismatches have negative scores and
-  thus need a signed score type ([\#1891](https://github.com/seqan/seqan3/pull/1891)).
+* Using an unsigned score type via the  `seqan3::align_cfg::score_type`configuration is prevented with a static assert,
```
Comment (marehr) [inline CHANGELOG.md:216]:
```suggestion
* Using an unsigned score type via the `seqan3::align_cfg::score_type` configuration is prevented with a static assert,
```

**Review by marehr — COMMENTED** [review]
thank you :)

File: CHANGELOG.md (hunk 3 above)
Comment (eseiler) [reply CHANGELOG.md:158]: Yes, I moved them up

File: CHANGELOG.md [hunk 5]
```diff
@@ -82,108 +90,146 @@ Note that 3.1.0 will be the first API stable release and interfaces in this rele
...
-### Alignment
+#### Alignment
 
 * When invoking the alignment algorithm with a user defined thread count using the `seqan3::align_cfg::parallel`
-  configuration element, `std::thread::hardware_concurrency()` many threads were always spawned. This is now fixed and
+  configuration element, all available threads were used. This is now fixed and
   only the specified number of threads will be spawned ([\#1854](https://github.com/seqan/seqan3/pull/1854)).
-* Using an unsigned `score_type` is prevented with a static assert, since gaps and mismatches have negative scores and
-  thus need a signed score type ([\#1891](https://github.com/seqan/seqan3/pull/1891)).
+* Using an unsigned score type via the  `seqan3::align_cfg::score_type` configuration is prevented with a static assert,
```
Comment (marehr) [inline CHANGELOG.md:211]:
```suggestion
* Using an unsigned score type via the `seqan3::align_cfg::score_type` configuration is prevented with a static assert,
```

**Review by marehr — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 6]
```diff
@@ -82,109 +90,147 @@ Note that 3.1.0 will be the first API stable release and interfaces in this rele
...
 * Removed `seqan3::bi_fm_index_cursor::to_rev_cursor()` and `seqan3::bi_fm_index::rev_cursor()`
   ([\#1892](https://github.com/seqan/seqan3/pull/1892)).
 
+Header Changes:
+
+```cpp
+#include <seqan3/search/{algorithm => }/search.hpp>
+#include <seqan3/{search/search_result_range.hpp => core/algorithm/algorithm_result_generator_range.hpp}>
+
+#include <seqan3/search/configuration/{mode.hpp => hit.hpp}>
```
Comment (smehringer) [inline CHANGELOG.md:200]:
```suggestion
#include <seqan3/{search/search_result_range.hpp => core/algorithm/algorithm_result_generator_range.hpp}>
#include <seqan3/search/configuration/{mode.hpp => hit.hpp}>
```
?

**Review by smehringer — APPROVED** [review]

File: CHANGELOG.md (hunk 6 above)
Comment (eseiler) [reply CHANGELOG.md:200]: There's always a new line if the subdirectory changes (`search/` and `search/configuration`)

Comment (smehringer) [reply CHANGELOG.md:200]: ah ok

**Pull request merged**